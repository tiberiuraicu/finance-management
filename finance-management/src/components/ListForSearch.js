import { TouchableOpacity, FlatList } from "react-native";

import Card from "./Card";
import CustomRow from "./CustomRow";
import CustomText from "./CustomText";

export const ListForSearch = (props) => {
  return (
    <FlatList
      data={props.data}
      keyboardShouldPersistTaps="always"
      renderItem={(itemData) => (
        <TouchableOpacity
          key={itemData.item.name}
          onPress={() => props.onPress(itemData.item.symbol)}
        >
          <Card>
            <CustomRow>
              <CustomText>{itemData.item.symbol}</CustomText>
              <CustomText>{itemData.item.name}</CustomText>
            </CustomRow>
          </Card>
        </TouchableOpacity>
      )}
      keyExtractor={(item, index) => index.toString()}
    />
  );
};
export default ListForSearch;
