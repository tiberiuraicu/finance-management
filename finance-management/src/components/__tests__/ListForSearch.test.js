import { ListForSearch } from "../ListForSearch";
import { shallow, mount } from "enzyme";

describe("ListFor search tests", () => {
  it("renders the component", () => {
    const wrapper = shallow(<ListForSearch />);
    expect(wrapper).toMatchSnapshot();
  });

  it("flatlist renders given items", () => {
    const wrapper = mount(
      <ListForSearch
        data={[
          { symbol: "TSLA", name: "Tesla" },
          { symbol: "MSFT", name: "Microsoft" },
        ]}
      />
    );
    const TouchableOpacities = wrapper.find("TouchableOpacity");
    expect(TouchableOpacities.length).toBe(2);
  });

  it("calls the onPress function with the right parameter", () => {
    const onPressMockFunction = jest.fn();
    const wrapper = mount(
      <ListForSearch
        data = {[
          { symbol: "TSLA", name: "Tesla" },
          { symbol: "MSFT", name: "Microsoft" },
        ]}
        onPress = {onPressMockFunction}
      />
    );

    const TouchableOpacity = wrapper.find("TouchableOpacity").at(0);
    TouchableOpacity.simulate("click");

    expect(onPressMockFunction).toBeCalledWith("TSLA");
  });
});
