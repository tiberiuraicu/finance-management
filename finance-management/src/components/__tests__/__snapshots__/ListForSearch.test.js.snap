// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListFor search tests renders the component 1`] = `ShallowWrapper {}`;
